### ISSUE TITLE
Allow devicelab A/B test to compare 2 local engines

### ISSUE THREAD

**Issue by liyuqian** [flutter/flutter#58726]
The current A/B test only tests the --local-engine=xxx and the default engine

The default engine and the local engine might be compiled with different settings (e.g., default engine probably has --lto while the local engine has --no-lto), so the comparison isn't completely controlled if we just want to test a local engine change.

A more fair comparison would be a local-to-local testing with the exact same compilation setting, but just with different engine code (say two git branches)

**Comment by liyuqian** [comment]
CC @flar. I just got some time today to put our discussions into a Github issue so we can track it.

**Comment by digiter** [comment]
Triage: is this implemented? @yjbanov

**Comment by flar** [comment]
Not implemented yet.

**Comment by digiter** [comment]
I thought https://github.com/flutter/flutter/pull/54494 is related?

**Comment by flar** [comment]
They are related. https://github.com/flutter/flutter/pull/54494 is about comparing the default engine to a local engine. This PR is about expanding that A/B mode to compare 2 different local engines rather than always assuming that you want to compare to the default engine.

**Comment by pedromassangocode** [comment]
Is this still valid?

Adding `engine` and `test` based on the title.

**Comment by liyuqian** [comment]
Yes, I believe it's still valid but we're not working on it yet.
